test/gtest/ucs/test_obj_size.cc

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2016.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include <common/test.h>
+#include <ucs/type/cpu_set.h>
+
+#include <ucp/core/ucp_ep.h>
+#include <ucp/core/ucp_request.h>
+#include <ucp/core/ucp_types.h>
+#include <uct/api/tl.h>
+#include <uct/base/uct_iface.h>
+#include <uct/ib/dc/base/dc_ep.h>
+#include <uct/ib/dc/base/dc_iface.h>
+#include <uct/ib/dc/verbs/dc_verbs.h>
+#include <uct/ib/rc/verbs/rc_verbs.h>
+#include <uct/ib/ud/base/ud_ep.h>
+#include <uct/ib/ud/verbs/ud_verbs.h>
+#include <uct/sm/self/self_ep.h>
+#include <uct/tcp/tcp.h>
+
+class test_obj_size : public ucs::test {
+};
+
+#if ENABLE_DEBUG_DATA
+#  define OBJ_SIZE(_rel, _dbg) _dbg
+#else
+#  define OBJ_SIZE(_rel, _dbg) _rel
+#endif
+
+#define EXPECTED_SIZE(_obj, _rel, _dbg) EXPECT_LE(sizeof(_obj), OBJ_SIZE(_rel, _dbg))
```

Comment:
i would make it EXPECT_EQ so we could also see optimizations and if there is optimization and later on a regression, it it would also be caught

Reply:
EXPECT_EQ may fail because it may have different sizes on IA/ARM platforms